pliers/transformers/api/base.py

```diff
@@ -0,0 +1,38 @@
+''' Base implementation for all API transformers. '''
+
+from pliers import config
+from pliers.transformers import Transformer
+from pliers.utils import isiterable, EnvironmentKeyMixin
+
+
+class APITransformer(Transformer, EnvironmentKeyMixin):
+
+    def _iterate(self, stims, validation='strict', *args, **kwargs):
+        # Check if we are trying to transform a large number of stimuli
+        if not config.get_option('allow_large_jobs'):
+            stims = list(stims)  # TODO: better way to do this?
```

Comment:
Hmm... I don't know that there's any way to check length without consuming a generator. We might have to initialize a counter that gets incremented each time through, and check that way. Otherwise we risk making the `use_generators` config option largely useless.